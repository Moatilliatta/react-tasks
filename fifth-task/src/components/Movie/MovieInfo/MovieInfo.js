import React from 'react';
import './MovieInfo.scss';

const MovieInfo = ({ movie, close }) => {
  return (
    <div className="MovieInfo">
      <div className="top-info">
        <span>netflix roulette</span>
        <label onClick={close}><img alt={movie.title} src={process.env.PUBLIC_URL+'search.png'} /></label>
      </div>
      <main>
        <img alt={movie.title} src={movie.imgSrc} />
        <section>
          <p className="title-genre search-section">
            <label>{movie.title} <span>{movie.rating}</span></label>
          </p>
          
          <p className="genre">
            <label>{movie.genre}</label>
          </p>

          <p className="release-duration">
            <label>{movie.releaseDate}</label><span>{movie.duration} min</span>
          </p>

          <p className="description">
            <label>{movie.description}</label>
          </p>
         </section>
      </main>
    </div>
  )
}

export default MovieInfo;