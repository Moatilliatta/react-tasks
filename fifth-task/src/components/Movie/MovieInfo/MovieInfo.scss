.MovieInfo {
  display: flex;
  flex-direction: column;
  margin: 0 4em;
  height: inherit;

  .top-info{
    display: flex;
    justify-content: space-between;
    margin: 2em 0;

    img {
      width: 2em;
    }
  }

  main {
    display: flex;
    flex-direction: row;

    img {
      height: 17em;
      margin-right: 3em;
    }

    section {
      text-transform: none;

      p {
        margin: 0 0 1em 0;
      }

      .title-genre {
        label {
          margin: 0;
          span {
                border: 1px solid white;
                border-radius: 1em;
                padding: 8px;
                font-size: .8em;
                color: lawngreen;
                width: 33px;
                display: inline-block;
          }
        }
      }

      .release-duration {
        display: flex;
        justify-content: space-between;
        width: 8em;
      }

      .description {
        overflow-y: auto;
        height: 7em;
      }
      .release-duration {
        color: #F65261;
        font-size: 1.5em;
      }
    }
  }
}