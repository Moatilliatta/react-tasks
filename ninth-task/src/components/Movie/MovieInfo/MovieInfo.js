import React from 'react';
import './MovieInfo.scss';
import { Img } from 'react-image';
import { getReleaseYear, backendHost } from '../../../utils/sharedFunctions';

const MovieInfo = ({ movie, close }) => {
  return (
    <div className="MovieInfo">
      <div className="top-info">
        <span>netflix roulette</span>
        <label onClick={close}>
          <img alt={movie.title} src={`${backendHost}/search.png`} />
        </label>
      </div>
      <main>
        <Img
          alt={movie.title}
          src={[
            movie.poster_path,
            `${backendHost}/no-image-available.jpg`
            ]}
          />
        <section>
          <p className="title-genre search-section">
            <label>{movie.title} <span>{movie.vote_average}</span></label>
          </p>
          <p className="genre">
            <label>{movie.genres.join(', ')}</label>
          </p>
          <p className="release-duration">
            <label>{getReleaseYear(movie.release_date)}</label><span>{movie.runtime} min</span>
          </p>
          <p className="description">
            <label>{movie.overview}</label>
          </p>
         </section>
      </main>
    </div>
  )
}

export default MovieInfo;