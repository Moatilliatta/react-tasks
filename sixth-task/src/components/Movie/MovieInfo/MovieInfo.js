import React from 'react';
import './MovieInfo.scss';
import { getReleaseYear } from '../../../utils/sharedFunctions';

const MovieInfo = ({ movie, close }) => {
  return (
    <div className="MovieInfo">
      <div className="top-info">
        <span>netflix roulette</span>
        <label onClick={close}>
          <img alt={movie.title} src={process.env.PUBLIC_URL+'search.png'} />
        </label>
      </div>
      <main>
        <img alt={movie.title} src={movie.poster_path} />
        <section>
          <p className="title-genre search-section">
            <label>{movie.title} <span>{movie.vote_average}</span></label>
          </p>
          <p className="genre">
            <label>{movie.genres.join(', ')}</label>
          </p>
          <p className="release-duration">
            <label>{getReleaseYear(movie.release_date)}</label><span>{movie.runtime} min</span>
          </p>
          <p className="description">
            <label>{movie.overview}</label>
          </p>
         </section>
      </main>
    </div>
  )
}

export default MovieInfo;