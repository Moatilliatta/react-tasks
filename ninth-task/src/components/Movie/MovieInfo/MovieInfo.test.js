import React from "react";
import MovieInfo from "./MovieInfo";
import renderer from "react-test-renderer";

describe("<MovieInfo />", () => {
  let movieData;

  beforeEach(() => {
    movieData = {
      title: "",
      poster_path: "",
      genres: [],
      release_date: "",
      overview: "",
    };
  });

  it("should set the movie info with movie object and close action", () => {
    const tree = renderer
      .create(<MovieInfo movie={movieData} close={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // A bit overkill, but just to know how to use inlineSnapshot.
  it("should show default no-image when not provided in movie object", () => {
    movieData.poster_path = "";
    const tree = renderer
      .create(<MovieInfo movie={movieData} close={() => {}} />)
      .toJSON();
    expect(tree).toMatchInlineSnapshot(`
      <div
        className="MovieInfo"
      >
        <div
          className="top-info"
        >
          <span>
            netflix roulette
          </span>
          <label
            onClick={[Function]}
          >
            <img
              alt=""
              src="//localhost/search.png"
            />
          </label>
        </div>
        <main>
          <section>
            <p
              className="title-genre search-section"
            >
              <label>
                
                 
                <span />
              </label>
            </p>
            <p
              className="genre"
            >
              <label>
                
              </label>
            </p>
            <p
              className="release-duration"
            >
              <label>
                NaN
              </label>
              <span>
                 min
              </span>
            </p>
            <p
              className="description"
            >
              <label>
                
              </label>
            </p>
          </section>
        </main>
      </div>
    `);
  });
});
