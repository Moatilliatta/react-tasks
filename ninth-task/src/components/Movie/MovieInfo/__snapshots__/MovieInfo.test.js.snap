// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfo /> should set the movie info with movie object and close action 1`] = `
<div
  className="MovieInfo"
>
  <div
    className="top-info"
  >
    <span>
      netflix roulette
    </span>
    <label
      onClick={[Function]}
    >
      <img
        alt=""
        src="//localhost/search.png"
      />
    </label>
  </div>
  <main>
    <section>
      <p
        className="title-genre search-section"
      >
        <label>
          
           
          <span />
        </label>
      </p>
      <p
        className="genre"
      >
        <label>
          
        </label>
      </p>
      <p
        className="release-duration"
      >
        <label>
          NaN
        </label>
        <span>
           min
        </span>
      </p>
      <p
        className="description"
      >
        <label>
          
        </label>
      </p>
    </section>
  </main>
</div>
`;
